[GlobalISel] Dump the machine function after each legalization iteration. NFC

davemgreen opened the pull request:
I find this useful to see the current state of the machine function as legalization continues. Otherwise it is difficult to view the whole thing as individual combines/legalizations are occurring.

arsenm commented on llvm/lib/CodeGen/GlobalISel/Legalizer.cpp:
```diff
@@ -223,6 +223,7 @@ Legalizer::legalizeMachineFunction(MachineFunction &MF, const LegalizerInfo &LI,
   SmallVector<MachineInstr *, 128> RetryList;
   do {
     LLVM_DEBUG(dbgs() << "=== New Iteration ===\n");
+    LLVM_DEBUG(MF.dump());
```
Can keep this in the one LLVM_DEBUG

aemerson replied:
Yeah this will be way too verbose, maybe just printing the block is likely to be enough to see what's going on most of the time.

Adding this guarded under a special debug flag like `-gisel-legalizer-print-after-iter` should be ok.

davemgreen replied:
I often use -print-after-all along with -debug, so it shouldn't be much more than that. It should be much less than double the existing debug output for legalization. I'd expect it shouldn't do too many iterations, maybe 1-3 is common? It's just something I find useful though, I'm happy to drop this if others disagree. I can just keep adding it locally when I need it.

aemerson replied:
I'd still prefer to have this opt-in since with -print-after-all I generally expect entire MFs to be printed after each full pass only.

davemgreen committed:
[GlobalISel] Dump the machine function after each legalization iteration. NFC

I find this useful to see the current state of the machine function as
legalization continues. Otherwise it is difficult to view the whole status
as individual combines/legalizations are occuring.
